File: src/Tldr.js
//@flow

/*******************************************************************************
 * Imports
 *******************************************************************************/

import type { History, Location } from 'history'

import type { Options as IndexOptions } from 'tldr/Index'
import type { Command } from 'tldr/Command'
import type { Page }    from 'tldr/Page'

/*******************************************************************************
 * Type Definitions
 *******************************************************************************/

type Debug = boolean

export type Parameters = {
  command:  Command;
  debug?:   Debug;
  history:  History;
  index:    IndexOptions;
  location: Location;
}

export type State = {
  params: Parameters;
  found?: boolean;
  page?:  Page;
}

File: src/tldr.ml
type package = {
  version:  string;
  revision: string;
}

type meta_state = {
  pkg: package;
  debug: bool;
}

type state = {
  meta: meta_state;
  window: Window.window;
  found: bool;
  current_index: Index.index;
  current_cmd: Index.command;
  current_page: Page.page;
}
